### ISSUE TITLE
[Forecasted Pipeline] Update consumeGrantModifications script

### ISSUE THREAD

**Issue by jeffsmohan** [usdigitalresponse/usdr-gost#3247]
### Why is this issue important?

We want forecasted grants available in our dataset (alongside active grants) so we can integrate them into the product for our users. 

### Current State

The `src/scripts/consumeGrantModifications.js` script is responsible for taking grant updated events from the grants-ingest pipeline and reflecting those grant updates in our postgres database. 

### Expected State

This script needs to be updated to correctly handle forecasted grants alongside normal grant opportunities. This includes handling new forecast-only fields, as well as handling nulls in grant-only fields. ([See here](https://github.com/usdigitalresponse/usdr-gost/issues/3212#issue-2372968120) for the field reference.)

### Implementation plan

**Blocked on #3212 and [grants-ingest#869
](https://github.com/usdigitalresponse/grants-ingest/issues/869) ** 

Update the `src/lib/grants-ingest.js` helper's `mapSourceDataToGrant` function to return a grant object with the following changes: 

- `close_date` should be set to `undefined` if not present (instead of `'2100-01-01'`)
- `estimated_award_date` should be set to the correct date value from the event, if present
- `estimated_project_start_date` should be set to the correct date value from the event, if present
- `estimated_synopsis_close_date` should be set to the correct date value from the event, if present
- `estimated_synopsis_close_date_explanation` should be set to the value from the event, or undefined
- `estimated_synopsis_post_date` should be set to the correct date value from the event, if present
- `fiscal_year` should be set to the correct date value from the event, if present
- `grantor_contact_name` should be set, if present
- `grantor_contact_phone_number` should be set, if present
- `is_forecasted` should be set to true if it's true on the event

**Comment by TylerHendrickson** [comment]
Sprinkling some documentation assets around:
- [Federal Grants data field mappings spreadsheet](https://docs.google.com/spreadsheets/d/10E3ydAADbGeOgrpNDjQmN5qBCIh569QsJwv__lgcep0/edit?usp=sharing)
- [LucidChart diagram (companion to the above) collaboration](https://lucid.app/lucidchart/5babf760-ba59-4f1b-85cb-9983ce37572c/edit?viewport_loc=-788%2C5%2C3531%2C2468%2C-lYf6udBqEj~&invitationId=inv_fe9d59e5-0765-4906-98e1-b460f7817e48)
  - [View-only diagram published 2024-07-29](https://lucid.app/lucidchart/5babf760-ba59-4f1b-85cb-9983ce37572c/view?page=-lYf6udBqEj~#)

**Comment by ClaireValdivia** [comment]
@masimons to review this to assess if this is a duplicate of PR that is currently in review (i.e. mapping logic)
